sim/druid/berserk.go

```diff
@@ -0,0 +1,53 @@
+package druid
+
+import (
+	"time"
+
+	"github.com/wowsims/wotlk/sim/core"
+	"github.com/wowsims/wotlk/sim/core/proto"
+)
+
+func (druid *Druid) registerBerserkCD() {
+	actionId := core.ActionID{SpellID: 50334}
+
+	glyphBonus := core.TernaryDuration(druid.HasMajorGlyph(proto.DruidMajorGlyph_GlyphOfBerserk), time.Second*5.0, 0.0)
+
+	druid.BerserkAura = druid.RegisterAura(core.Aura{
+		Label:    "Berserk",
+		ActionID: actionId,
+		Duration: (time.Second * 15) + glyphBonus,
+		OnGain: func(aura *core.Aura, sim *core.Simulation) {
+			if druid.InForm(Cat) {
+				druid.PseudoStats.CostMultiplier /= 2.0
+			}
+		},
+		OnExpire: func(aura *core.Aura, sim *core.Simulation) {
+			if druid.InForm(Cat) {
+				druid.PseudoStats.CostMultiplier *= 2.0
+			}
+		},
+	})
+
+	if !druid.Talents.Berserk {

```

Comment:
This check can be placed at the start of the function. It's better to avoid registering auras when they can't be used, because its better for performance. Also, note that many of the getter functions for auras, such as `IsActive` and `GetStacks` work fine even if the aura is `nil`.

Reply:
interesting... did not realize thats how go works, definitely had it there cause i assumed it'd segfault otherwise ha, will change

Reply:
A null dereference is still a segfault/panic in Go. However, Go allows methods to check themselves for `nil`, which we have done for these functions for convenience: https://github.com/wowsims/wotlk/blob/master/sim/core/aura.go#L131